schema_utils.py

```diff
@@ -387,3 +411,89 @@ def is_valid_email(obj):
             bool. Whether the given object is a valid email.
         """
         return bool(re.search(r'^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$', obj))
+
+    @staticmethod
+    def is_valid_expression(obj, algebraic=True):
+        """Checks if the given  obj (a string) represents a valid algebraic or
+        numeric expression.
+
+        Args:
+            obj: str. The given expression.
+            algebraic: bool. True if the given expression is algebraic
+            else numeric.
+
+        Returns:
+            bool. Whether the given object is a valid expression.
+        """
+        valid_expression = True
+        valid_characters = '({[]})+-/*^.'
+
+        # Expression should not contain invalid characters.
+        for character in obj:
+            if character not in valid_characters and not bool(
+                    re.match(r'[a-zA-Z0-9]', character)):
+                valid_expression = False
+                break
+
+        # Algebraic expressions should contain at least one latin letter and
+        # numeric expressions should contain none.
+        if algebraic:
+            valid_expression &= bool(re.search(r'[a-zA-Z]+', obj))
+        else:
+            valid_expression &= not bool(re.search(r'[a-zA-Z]+', obj))
+
+        # Expression should contain a balanced bracket sequence.
+        valid_expression &= contains_balanced_brackets(obj)
+
+        # Expression should not contain multiple operators consecutively.
+        valid_operators = valid_characters[6:]
```

Comment:
No magic numbers in the code. Perhaps have separate constants for brackets/non-brackets chars.

Reply:
Done.